src/js/base/ProgressCircle.tsx

```diff
@@ -0,0 +1,223 @@
+/**
+ * Calculate the circumference of the circle taking into account the stroke width
+ *
+ * @param size - The total size of the progress circle in pixels
+ * @returns The circumference of the circle in pixels
+ */
+function getCircumference(size: number): number {
+    return getRadius(size) * Math.PI * 2;
+}
+
+/**
+ * Calculate stroke offset needed to display the correct amount of progress
+ *
+ * @param size - The total size of the progress circle in pixels
+ * @param progress - the amount of progress from 0 to 1
+ * @returns the stroke offset in pixels
+ */
+function getStrokeDashoffset({ progress, size }: { progress: number; size: number }): number {
+    return (1 - progress) * getCircumference(size);
+}
+
+const rotate = keyframes`
+        0% {
+            transform: rotate(0deg);
+        }
+        50% { 
+            transform: rotate(180deg);
+        }
+        100% {
+            transform: rotate(360deg);
+        }
+    `;
+
+type ProgressCircleBaseProps = {
+    size: number;
+};
+
+/**
+ * The root progress circle component.
+ *
+ * @param size - The total size of the progress circle in pixels
+ */
+const StyledProgressCircle = styled.svg<ProgressCircleBaseProps>`
+    width: ${props => props.size}px;
+    height: ${props => props.size}px;
+    transform: rotate(-90deg);
+`;
+
+/**
+ * Circle component containing shared ProgressCircle styles
+ *
+ * @param size - The total size of the progress circle in pixels
+ * @returns The ProgressCircle indicator
+ */
+
+const ProgressCircleBase = styled.circle<ProgressCircleBaseProps>`
+    cx: ${props => props.size / 2}px;
+    cy: ${props => props.size / 2}px;
+    r: ${props => getRadius(props.size)}px;
+    fill: transparent;
+    stroke-width: ${props => getStrokeWidth(props.size)}px;
+    stroke: ${getColor};
+    stroke-dasharray: ${props => getCircumference(props.size)}px;
+    transition: stroke-dashoffset 1s, stroke 1s;
+`;
+
+type ProgressCircleIndicatorProps = ProgressCircleBaseProps & {
+    progress: number;
+};
+
+/**
+ * Indicator for the amount of progress that has been made.
+ *
+ * @param size - The total size of the progress circle in pixels
+ * @param progress - The amount of progress that has been made from 0 to 1
+ * @returns The ProgressCircle indicator
+ */
+const ProgressCircleIndicator = styled(ProgressCircleBase)<ProgressCircleIndicatorProps>`
+    stroke-dashoffset: ${getStrokeDashoffset}px;
+`;
+
+type ProgressCircleTrackProps = ProgressCircleBaseProps & {
+    state: JobState;
+};
+
+/**
+ * The indeterminate loader and track for the determinate progress circle.
+ *
+ * @param size - The total size of the progress circle in pixels
+ * @param state - The current state of the job or task
+ * @returns The progress circle track
+ */
+const ProgressCircleTrack = styled(ProgressCircleBase)<ProgressCircleTrackProps>`
+    stroke-dashoffset: ${props =>
+        props.state === "waiting" ? `${getStrokeDashoffset({ progress: 0.75, size: props.size })}px` : "0px"};
```

Comment:
Could this be pushed into `getStrokeDashOffset`?

Reply:
The state check? I would prefer not to since the progress indicator uses the same function to calculate `stroke-dashoffset` so it can display the jobs progress. I could do it, but would have to account for it in ways that, imo make the component logic less clear.

Reply:
I agree.